plugin/src/main/kotlin/io/github/ttypic/swiftklib/gradle/task/CompileSwiftTask.kt

```diff
@@ -161,7 +164,7 @@ open class CompileSwiftTask @Inject constructor(
             staticLibraries = ${libPath.name}
             libraryPaths = ${libPath.parentFile.absolutePath}
 
-            linkerOpts = -L/usr/lib/swift -${compileTarget.linkerMinIosVersionName()} $minIos.0 -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/${compileTarget.os()}
+            linkerOpts = -L/usr/lib/swift -${compileTarget.linkerMinIosVersionName()} ${minOs(compileTarget, minIos, minMacosx)}.0 -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/${compileTarget.os()}
```

Comment:
@davidtaylor-juul do you happen to know where we can find official documentation of what the min OS version argument should look like?

Reply:
The nearest I could find is in the linker docs (`man ld`) which has:
```
     -platform_version platform min_version sdk_version
             This is set to indicate the platform, oldest supported version of that platform that output is to be used on, and the SDK that
             the output was built against.  platform is a numeric value as defined in <mach-o/loader.h>, or it may be one of the following
             strings:
             • macos
             • ios
             • tvos
             • watchos
             • bridgeos
             • mac-catalyst
             • ios-simulator
             • tvos-simulator
             • watchos-simulator
             • driverkit
             Specifying a newer min or SDK version enables the linker to assume features of that OS or SDK in the output file. The format of
             min_version and sdk_version is a version number such as 10.13 or 10.14

     -macos_version_min version
             This is set to indicate the oldest macOS version that that the output is to be used on.  Specifying a later version enables the
             linker to assumes features of that OS in the output file.  The format of version is a macOS version number such as 10.9 or
             10.14

     -ios_version_min version
             This is set to indicate the oldest iOS version that that the output is to be used on.  Specifying a later version enables the
             linker to assumes features of that OS in the output file.  The format of version is an iOS version number such as 3.1 or 4.0
```

Note that _probably_ some of the Apple targets that are iOS-like (watch and tv maybe?) might work with the `ios_version_min` flag, so I'm not sure if we would need to support the `-platform_version` syntax.

Reply:
For `linkerMinIosVersionName`, https://opensource.apple.com/source/cctools/cctools-949.0.1/misc/libtool.c.auto.html hints at what min version name parameters are available. I've adjusted them accordingly.